atef/widgets/config/data_active.py

```diff
@@ -880,13 +881,12 @@ def confirm_signal(self) -> None:
         try:
             sig.wait_for_connection()
```

Comment:
This is another spot where blocking the GUI thread will easily happen (connect -> block GUI thread -> timeout and raise). This should be moved to a background thread.